# Pandas - Ejercicios dataset Titanic



- - -

In [3]:
import pandas as pd

In [4]:
df = pd.read_csv("https://raw.githubusercontent.com/4data-lab/datasets/master/titanic.csv")

#### Columnas
survived (Sobrevivió)  
pclass (Clase de números)  
sex (Sexo)  
age (Edad)  
sibsp (Número de hermanos/cónyuges abordo)  
parch (Número de padres/hijos)   
fare (Tarifa)  
embarked (Embarcado)  
class (Clase en texto)  
adult_male (Hombre Adulto)  
deck (Cubierta)  
embark_town (Pueblo de embarcación)  
alive (Vivo)  
alone (Solo)


- - -
1 - Muestra solo los primeros 5 registros

In [ ]:
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

   adult_male deck  embark_town alive  alone  
0        True  NaN  Southampton    no  False  
1       False    C    Cherbourg   yes  False  
2       False  NaN  Southampton   yes   True  
3       False    C  Southampton   yes  False  
4        True  NaN  Southampton    no   True

- - -
2 - Obtén cuales son las distintas clases de viaje de este dataframe

`array(['Third', 'First', 'Second'], dtype=object)`

In [ ]:
df.dtypes # 'Third', 'First', 'Second' son valores de la columa "class" que vemos que es de tipo "object"

survived         int64
pclass           int64
sex             object
age            float64
sibsp            int64
parch            int64
fare           float64
embarked        object
class           object
adult_male        bool
deck            object
embark_town     object
alive           object
alone             bool
dtype: object

In [ ]:
df["class"].unique() # Devuelve los valores únicos de la columna "class" y su tipo. ## ES LO PEDIDO EN EJERCICIO 2

array(['Third', 'First', 'Second'], dtype=object)

- - -
3 - Obten el valor máximo de edad en el dataframe

`Out: 80.0`

In [ ]:
df["age"].max()

80.0

- - -
4 - Obten cúantos valores distintos hay en la columna de Pueblo de embarcación

`Out: 3`

In [ ]:
df["embarked"].count() # Cuantos valores embarked hay (o sea valores que no sean missing values)

889

In [ ]:
df["embarked"].unique() # Cuales son los valores de embarked úncios? Vemos que son 3: S, C, Q

array(['S', 'C', 'Q', nan], dtype=object)

In [ ]:
df["embarked"].nunique() # Cuantos valores de embarked diferentes hay?  ## ES LO PEDIDO EN EJERCICIO 4

3

- - -
5 - Muestra solo las primeras cinco edades más frencuentes

`Out:
24.0    30
22.0    27
18.0    26
19.0    25
30.0    25
Name: age, dtype: int64`

In [ ]:
df["age"].value_counts() # frecuencia absoluta de todos los valores

24.00    30
22.00    27
18.00    26
19.00    25
28.00    25
         ..
36.50     1
55.50     1
0.92      1
23.50     1
74.00     1
Name: age, Length: 88, dtype: int64

In [ ]:
df["age"].value_counts().head() # frecuencia absoluta de solo los 5 primeros valores

24.0    30
22.0    27
18.0    26
19.0    25
28.0    25
Name: age, dtype: int64

In [ ]:
list(df["age"].value_counts().head()) # Para tener la lista

[30, 27, 26, 25, 25]

In [ ]:
 # Para tener el Out pedido en el Ejercicio
string=""
for i in list(df["age"].value_counts().head()):
  string += str(i)+ ".0 "
print("Out: " + string + " Name: age, dtype: int64")

Out: 30.0 27.0 26.0 25.0 25.0  Name: age, dtype:float64


- - -
6 - Muestra los registros que cumplan la condición de disponer el valor de cubierta ("deck") "F"

In [ ]:
df[df["deck"] == "F"]

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
66          1       2  female  29.0      0      0  10.5000        S  Second   
75          0       3    male  25.0      0      0   7.6500        S   Third   
128         1       3  female   NaN      1      1  22.3583        C   Third   
148         0       2    male  36.5      0      2  26.0000        S  Second   
183         1       2    male   1.0      2      1  39.0000        S  Second   
193         1       2    male   3.0      1      1  26.0000        S  Second   
340         1       2    male   2.0      1      1  26.0000        S  Second   
345         1       2  female  24.0      0      0  13.0000        S  Second   
516         1       2  female  34.0      0      0  10.5000        S  Second   
618         1       2  female   4.0      2      1  39.0000        S  Second   
699         0       3    male  42.0      0      0   7.6500        S   Third   
715         0       3    male  19.0      0      0   7.6500        S   Third   
776         0       3    male   NaN      0      0   7.7500        Q   Third   

     adult_male deck  embark_town alive  alone  
66        False    F  Southampton   yes   True  
75         True    F  Southampton    no   True  
128       False    F    Cherbourg   yes  False  
148        True    F  Southampton    no  False  
183       False    F  Southampton   yes  False  
193       False    F  Southampton   yes  False  
340       False    F  Southampton   yes  False  
345       False    F  Southampton   yes   True  
516       False    F  Southampton   yes   True  
618       False    F  Southampton   yes  False  
699        True    F  Southampton    no   True  
715        True    F  Southampton    no   True  
776        True    F   Queenstown    no   True

In [ ]:
df["deck"] == "F"

0      False
1      False
2      False
3      False
4      False
       ...  
886    False
887    False
888    False
889    False
890    False
Name: deck, Length: 891, dtype: bool

- - -
7 - Muestra los registros que cumplan la condición de disponer más de 47 años y menos de 49 años

In [ ]:
df[(df["age"] > 47) & (df["age"] < 49)] #suponemos que el ejercicio pide más estricatmente y menos estrictamente

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
460         1       1    male  48.0      0      0  26.5500        S   First   
463         0       2    male  48.0      0      0  13.0000        S  Second   
556         1       1  female  48.0      1      0  39.6000        C   First   
645         1       1    male  48.0      1      0  76.7292        C   First   
712         1       1    male  48.0      1      0  52.0000        S   First   
736         0       3  female  48.0      1      3  34.3750        S   Third   
754         1       2  female  48.0      1      2  65.0000        S  Second   
771         0       3    male  48.0      0      0   7.8542        S   Third   
862         1       1  female  48.0      0      0  25.9292        S   First   

     adult_male deck  embark_town alive  alone  
460        True    E  Southampton   yes   True  
463        True  NaN  Southampton    no   True  
556       False    A    Cherbourg   yes  False  
645        True    D    Cherbourg   yes  False  
712        True    C  Southampton   yes  False  
736       False  NaN  Southampton    no  False  
754       False  NaN  Southampton   yes  False  
771        True  NaN  Southampton    no   True  
862       False    D  Southampton   yes   True

In [ ]:
 df[df["age"] == 48] # Relamente lo anterior es mismo que decir = 48

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
460         1       1    male  48.0      0      0  26.5500        S   First   
463         0       2    male  48.0      0      0  13.0000        S  Second   
556         1       1  female  48.0      1      0  39.6000        C   First   
645         1       1    male  48.0      1      0  76.7292        C   First   
712         1       1    male  48.0      1      0  52.0000        S   First   
736         0       3  female  48.0      1      3  34.3750        S   Third   
754         1       2  female  48.0      1      2  65.0000        S  Second   
771         0       3    male  48.0      0      0   7.8542        S   Third   
862         1       1  female  48.0      0      0  25.9292        S   First   

     adult_male deck  embark_town alive  alone  
460        True    E  Southampton   yes   True  
463        True  NaN  Southampton    no   True  
556       False    A    Cherbourg   yes  False  
645        True    D    Cherbourg   yes  False  
712        True    C  Southampton   yes  False  
736       False  NaN  Southampton    no  False  
754       False  NaN  Southampton   yes  False  
771        True  NaN  Southampton    no   True  
862       False    D  Southampton   yes   True

In [ ]:
df[(df["age"] >= 47) & (df["age"] <= 49)] # Si no queremos hacer la condición de estrictamente menor y estrictamente mayor, sino que también aceptamos los iguales sería así

survived  pclass     sex   age  sibsp  parch      fare embarked   class  \
52          1       1  female  49.0      1      0   76.7292        C   First   
110         0       1    male  47.0      0      0   52.0000        S   First   
132         0       3  female  47.0      1      0   14.5000        S   Third   
453         1       1    male  49.0      1      0   89.1042        C   First   
460         1       1    male  48.0      0      0   26.5500        S   First   
462         0       1    male  47.0      0      0   38.5000        S   First   
463         0       2    male  48.0      0      0   13.0000        S  Second   
515         0       1    male  47.0      0      0   34.0208        S   First   
556         1       1  female  48.0      1      0   39.6000        C   First   
586         0       2    male  47.0      0      0   15.0000        S  Second   
592         0       3    male  47.0      0      0    7.2500        S   Third   
597         0       3    male  49.0      0      0    0.0000        S   Third   
599         1       1    male  49.0      1      0   56.9292        C   First   
645         1       1    male  48.0      1      0   76.7292        C   First   
662         0       1    male  47.0      0      0   25.5875        S   First   
698         0       1    male  49.0      1      1  110.8833        C   First   
712         1       1    male  48.0      1      0   52.0000        S   First   
736         0       3  female  48.0      1      3   34.3750        S   Third   
754         1       2  female  48.0      1      2   65.0000        S  Second   
771         0       3    male  48.0      0      0    7.8542        S   Third   
796         1       1  female  49.0      0      0   25.9292        S   First   
862         1       1  female  48.0      0      0   25.9292        S   First   
871         1       1  female  47.0      1      1   52.5542        S   First   
873         0       3    male  47.0      0      0    9.0000        S   Third   

     adult_male deck  embark_town alive  alone  
52        False    D    Cherbourg   yes  False  
110        True    C  Southampton    no   True  
132       False  NaN  Southampton    no  False  
453        True    C    Cherbourg   yes  False  
460        True    E  Southampton   yes   True  
462        True    E  Southampton    no   True  
463        True  NaN  Southampton    no   True  
515        True    D  Southampton    no   True  
556       False    A    Cherbourg   yes  False  
586        True  NaN  Southampton    no   True  
592        True  NaN  Southampton    no   True  
597        True  NaN  Southampton    no   True  
599        True    A    Cherbourg   yes  False  
645        True    D    Cherbourg   yes  False  
662        True    E  Southampton    no   True  
698        True    C    Cherbourg    no  False  
712        True    C  Southampton   yes  False  
736       False  NaN  Southampton    no  False  
754       False  NaN  Southampton   yes  False  
771        True  NaN  Southampton    no   True  
796       False    D  Southampton   yes   True  
862       False    D  Southampton   yes   True  
871       False    D  Southampton   yes  False  
873        True  NaN  Southampton    no   True

- - -
8 - Obten los nombres de los pueblos de embarcación de los pasajeros de la cubierta "A"

`Out:
23     Southampton
96       Cherbourg
174      Cherbourg
185    Southampton
209      Cherbourg
284    Southampton
445    Southampton
475    Southampton
556      Cherbourg
583      Cherbourg
599      Cherbourg
630    Southampton
647      Cherbourg
806    Southampton
867    Southampton
Name: embark_town, dtype: object`

In [ ]:
df[df["deck"].isin(["A"])] # filtra el dataframe por los que tienen exactamente la palabra "A" en el campo "deck"

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
23          1       1    male  28.0      0      0  35.5000        S  First   
96          0       1    male  71.0      0      0  34.6542        C  First   
174         0       1    male  56.0      0      0  30.6958        C  First   
185         0       1    male   NaN      0      0  50.0000        S  First   
209         1       1    male  40.0      0      0  31.0000        C  First   
284         0       1    male   NaN      0      0  26.0000        S  First   
445         1       1    male   4.0      0      2  81.8583        S  First   
475         0       1    male   NaN      0      0  52.0000        S  First   
556         1       1  female  48.0      1      0  39.6000        C  First   
583         0       1    male  36.0      0      0  40.1250        C  First   
599         1       1    male  49.0      1      0  56.9292        C  First   
630         1       1    male  80.0      0      0  30.0000        S  First   
647         1       1    male  56.0      0      0  35.5000        C  First   
806         0       1    male  39.0      0      0   0.0000        S  First   
867         0       1    male  31.0      0      0  50.4958        S  First   

     adult_male deck  embark_town alive  alone  
23         True    A  Southampton   yes   True  
96         True    A    Cherbourg    no   True  
174        True    A    Cherbourg    no   True  
185        True    A  Southampton    no   True  
209        True    A    Cherbourg   yes   True  
284        True    A  Southampton    no   True  
445       False    A  Southampton   yes  False  
475        True    A  Southampton    no   True  
556       False    A    Cherbourg   yes  False  
583        True    A    Cherbourg    no   True  
599        True    A    Cherbourg   yes  False  
630        True    A  Southampton   yes   True  
647        True    A    Cherbourg   yes   True  
806        True    A  Southampton    no   True  
867        True    A  Southampton    no   True

In [ ]:
df[df["deck"].isin(["A"])]["embark_town"] # Para tener la columna de los pueblos de embarcación

23     Southampton
96       Cherbourg
174      Cherbourg
185    Southampton
209      Cherbourg
284    Southampton
445    Southampton
475    Southampton
556      Cherbourg
583      Cherbourg
599      Cherbourg
630    Southampton
647      Cherbourg
806    Southampton
867    Southampton
Name: embark_town, dtype: object

- - -

9 - Muestra la media de los valores del dataframe agrupando por sexo

In [ ]:
df.groupby("sex").mean()

<ipython-input-76-4636136becc0>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby("sex").mean()


survived    pclass        age     sibsp     parch       fare  \
sex                                                                    
female  0.742038  2.159236  27.915709  0.694268  0.649682  44.479818   
male    0.188908  2.389948  30.726645  0.429809  0.235702  25.523893   

        adult_male     alone  
sex                           
female    0.000000  0.401274  
male      0.930676  0.712305

- - -
10 - Define una función que capitalice strings y con pandas aplicala a la columna sexo

`Entrada : hola mundo`
`Salida: Hola mundo`
`Pista: capitalize()`

In [ ]:
def capital(s):
  return(s.capitalize())
df["sex"].apply(capital)

0        Male
1      Female
2      Female
3      Female
4        Male
        ...  
886      Male
887    Female
888    Female
889      Male
890      Male
Name: sex, Length: 891, dtype: object

11 - Reemplaza todos los NaN de la columna deck por la palabra 'Unknown'

In [ ]:
df["deck"].fillna("Unknown")

0      Unknown
1            C
2      Unknown
3            C
4      Unknown
        ...   
886    Unknown
887          B
888    Unknown
889          C
890    Unknown
Name: deck, Length: 891, dtype: object

12 - ¿Cuál es el precio promedio del billete que pagaron los pasajeros por cada clase (primera, segunda, tercera)?

`Out: First     84.154687
Second    20.662183
Third     13.675550 `

In [ ]:
df.groupby("class").mean().drop(["survived","pclass","age", "sibsp","parch","adult_male","alone"], axis=1)

<ipython-input-99-3a71af035abf>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby("class").mean().drop(["survived","pclass","age", "sibsp","parch","adult_male","alone"], axis=1)


fare
class            
First   84.154687
Second  20.662183
Third   13.675550

- - -

13 - ¿Cuántos supervivientes están todavía vivos?


 `Out: 342`

In [ ]:
df["survived"].sum() # Dado que 1=sobrevivió y 0=no sobrevivió, si sumamos los valores de esa columna sabremos cuantos sobrevivieron

342

In [ ]:
df["survived"].fillna(0).sum() # Hacemos lo mismo suponiendo que los que no sabemos si sobrevivieron, murieron

342

In [ ]:
df["survived"].fillna(1).sum() # Hacemos lo mismo suponiendo que los que no sabemos si sobrevivieron, sobrevivieron

342

- - -

14 - ¿Cuál fue el precio del billete de segunda clase más caro? ¿Y el más barato?


`Out : Caro: 73.5 Barato: 0.0 `

In [ ]:
df[df["class"].isin(["Second"])] # filtra el dataframe por los que tienen exactamente la palabra "Second" en el campo "class"

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
9           1       2  female  14.0      1      0  30.0708        C  Second   
15          1       2  female  55.0      0      0  16.0000        S  Second   
17          1       2    male   NaN      0      0  13.0000        S  Second   
20          0       2    male  35.0      0      0  26.0000        S  Second   
21          1       2    male  34.0      0      0  13.0000        S  Second   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
866         1       2  female  27.0      1      0  13.8583        C  Second   
874         1       2  female  28.0      1      0  24.0000        C  Second   
880         1       2  female  25.0      0      1  26.0000        S  Second   
883         0       2    male  28.0      0      0  10.5000        S  Second   
886         0       2    male  27.0      0      0  13.0000        S  Second   

     adult_male deck  embark_town alive  alone  
9         False  NaN    Cherbourg   yes  False  
15        False  NaN  Southampton   yes   True  
17         True  NaN  Southampton   yes   True  
20         True  NaN  Southampton    no   True  
21         True    D  Southampton   yes   True  
..          ...  ...          ...   ...    ...  
866       False  NaN    Cherbourg   yes  False  
874       False  NaN    Cherbourg   yes  False  
880       False  NaN  Southampton   yes  False  
883        True  NaN  Southampton    no   True  
886        True  NaN  Southampton    no   True  

[184 rows x 14 columns]

In [ ]:
#seleccionamos solo valores de precio billete
select = df[df["class"].isin(["Second"])].drop(["class","sex","survived","pclass","age", "sibsp","parch","adult_male","alone", "embarked",	"deck",	"embark_town","alive"], axis=1)
select

fare
9    30.0708
15   16.0000
17   13.0000
20   26.0000
21   13.0000
..       ...
866  13.8583
874  24.0000
880  26.0000
883  10.5000
886  13.0000

[184 rows x 1 columns]

In [ ]:
# Encontramos el máximo de allí
select.max()

fare    73.5
dtype: float64

In [ ]:
# Encontramos el mínimo de allí
select.min()

fare    0.0
dtype: float64

- - -

15- Elimina el tercer registro del dataframe

- - -

In [ ]:
df.drop(2) # Quitamos el tercer registro (fila 2)

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
5           0       3    male   NaN      0      0   8.4583        Q   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

     adult_male deck  embark_town alive  alone  
0          True  NaN  Southampton    no  False  
1         False    C    Cherbourg   yes  False  
3         False    C  Southampton   yes  False  
4          True  NaN  Southampton    no   True  
5          True  NaN   Queenstown    no   True  
..          ...  ...          ...   ...    ...  
886        True  NaN  Southampton    no   True  
887       False    B  Southampton   yes   True  
888       False  NaN  Southampton    no  False  
889        True    C    Cherbourg   yes   True  
890        True  NaN   Queenstown    no   True  

[890 rows x 14 columns]

In [ ]:
df.drop(3) # Quitamos la fila 3

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
4           0       3    male  35.0      0      0   8.0500        S   Third   
5           0       3    male   NaN      0      0   8.4583        Q   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

     adult_male deck  embark_town alive  alone  
0          True  NaN  Southampton    no  False  
1         False    C    Cherbourg   yes  False  
2         False  NaN  Southampton   yes   True  
4          True  NaN  Southampton    no   True  
5          True  NaN   Queenstown    no   True  
..          ...  ...          ...   ...    ...  
886        True  NaN  Southampton    no   True  
887       False    B  Southampton   yes   True  
888       False  NaN  Southampton    no  False  
889        True    C    Cherbourg   yes   True  
890        True  NaN   Queenstown    no   True  

[890 rows x 14 columns]

16- Añade un nuevo registro al dataframe de un hombre que embarcó desde el puerto de Barcelona. Puedes inventarte el resto de campos.

In [ ]:
nueva_fila = ["1", "3", "male", "28", "3", "2", "7.7500", "S", "Third", True, "C", "Barcelona", "yes", False]
#if not ( df.iloc[[1]].all() == nueva_fila ):
#  df.loc[len(df)] = nueva_fila
df.loc[len(df)] = nueva_fila
df.tail()

survived pclass   sex   age sibsp parch    fare embarked  class  \
889        1      1  male  26.0     0     0    30.0        C  First   
890        0      3  male  32.0     0     0    7.75        Q  Third   
891        1      3  male    28     3     2  7.7500        S  Third   
892        1      3  male    28     3     2  7.7500        S  Third   
893        1      3  male    28     3     2  7.7500        S  Third   

     adult_male deck embark_town alive  alone  
889        True    C   Cherbourg   yes   True  
890        True  NaN  Queenstown    no   True  
891        True    C   Barcelona   yes  False  
892        True    C   Barcelona   yes  False  
893        True    C   Barcelona   yes  False

- - -

17- Ordena el dataframe según la edad de los registros.

---



In [5]:
df.sort_values(by='age')

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
803         1       3    male  0.42      0      1   8.5167        C   Third   
755         1       2    male  0.67      1      1  14.5000        S  Second   
644         1       3  female  0.75      2      1  19.2583        C   Third   
469         1       3  female  0.75      2      1  19.2583        C   Third   
78          1       2    male  0.83      0      2  29.0000        S  Second   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
859         0       3    male   NaN      0      0   7.2292        C   Third   
863         0       3  female   NaN      8      2  69.5500        S   Third   
868         0       3    male   NaN      0      0   9.5000        S   Third   
878         0       3    male   NaN      0      0   7.8958        S   Third   
888         0       3  female   NaN      1      2  23.4500        S   Third   

     adult_male deck  embark_town alive  alone  
803       False  NaN    Cherbourg   yes  False  
755       False  NaN  Southampton   yes  False  
644       False  NaN    Cherbourg   yes  False  
469       False  NaN    Cherbourg   yes  False  
78        False  NaN  Southampton   yes  False  
..          ...  ...          ...   ...    ...  
859        True  NaN    Cherbourg    no   True  
863       False  NaN  Southampton    no  False  
868        True  NaN  Southampton    no   True  
878        True  NaN  Southampton    no   True  
888       False  NaN  Southampton    no  False  

[891 rows x 14 columns]

18 - Guarda el dataframe como un fichero de Excel.

In [12]:
from google.colab import drive
drive.mount('/content/drive')
df.to_excel("/content/drive/MyDrive/Titanic_Albert_Ripoll.xlsx", index=False, sheet_name="Hoja1")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


- - -

# Pandas - Ejercicios dataset Youtube

1- Descárgate el dataset de YouTube del eStudy y cárgalo en un DataFrame.

2- Muestra los 3 primeros registros.

3- ¿Cuál es el promedio de suscriptores que tiene cada canal?

4- ¿Qué canal tiene el nombre más largo?

5- ¿Qué porcentaje de canales son de marcas?

6- ¿Cuál es la categoría menos popular? ¿Y la más popular?

7- ¿Cuántos canales Educativos tienen el Inglés como lengua principal?

8- Crea una nueva columna en el dataset con el continente de cada canal.

9- Enumera el promedio de suscriptores por continente.

10- Crea un DataFrame independiente con la información de los canales de YouTube de Lady Gaga y La Salle Campus Barcelona que tenga como datos el nombre del canal, los suscriptores y la categoría.

11- Une el DataFrame creado en el punto 10 con el DataFrame original en un nuevo DataFrame.

12 - Modifica los nombres de los canales para que no haya espacio en ellos. Para ello, reemplaza todos los espacios por el carácter "_".